src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,85 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    static final int DEFAULT_CAPACITY = 16;
+    static final float LOAD_FACTOR = 0.75f;
+    private int size;
+    private Node<K, V>[] table;
+
+    public MyHashMap() {
+        table = new Node[DEFAULT_CAPACITY];
+    }
 
     @Override
     public void put(K key, V value) {
-
+        if (size > table.length * LOAD_FACTOR) {
+            ensureCapacity();
+        }
+        int keyHash = hash(key);
+        int bucketIndex = keyHash % table.length;
+        Node<K, V> currentNode = table[bucketIndex];
+        if (table[bucketIndex] == null) {
+            table[bucketIndex] = new Node<>(key, value, null);
+        }
+        while (currentNode != null) {
+            if (Objects.equals(currentNode.key, key)) {
+                currentNode.value = value;
+                return;
+            }
+            if (currentNode.next == null) {
+                currentNode.next = new Node<>(key, value, null);
+                break;
+            }
+            currentNode = currentNode.next;
+        }
+        size++;
     }
 
     @Override
     public V getValue(K key) {
+        int bucketIndex = hash(key) % table.length;
+        Node<K, V> currentNode = table[bucketIndex];
+        while (currentNode != null) {
+            if (Objects.equals(currentNode.key, key)) {
+                return currentNode.value;
+            }
+            currentNode = currentNode.next;
+        }
         return null;
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    private void ensureCapacity() {
+        Node<K, V> [] oldTable = table;
+        size = 0;
+        table = new Node[oldTable.length << 1];
+        for (Node<K, V> currentNode: oldTable) {
+            while (currentNode != null) {
+                put(currentNode.key, currentNode.value);
+                currentNode = currentNode.next;
+            }
+        }
+    }
+
+    private int hash(K key) {
+        return key == null ? 0 : Math.abs(key.hashCode() * 31);
```

Comment:
```suggestion
        return key == null ? 0 : Math.abs(key.hashCode());
```
why 31?

Reply:
to get hashcode even more unique, just in case :D